aten/src/ATen/templates/TypeExtensionBackend.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include <ATen/TypeDefault.h>
+
+namespace at {
+
+struct CAFFE2_API ${Type} : public TypeDefault {
+  explicit ${Type}();
+
+  template <typename FnPtr>
```

Comment:
A note here explaining the static member variable relationship to FnPtr typename would be appreciated!